modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Sequence.scala

```diff
@@ -57,27 +59,43 @@ object Sequence {
     val next: Option[Zipper] =
       focus.next match {
         // Step completed
-        case None      =>
-          pending match {
+        case None      => {
+          remaining match {
             case Nil             => None
             case stepp :: stepps =>
               (Step.Zipper.currentify(stepp) |@| focus.uncurrentify) (
-                (curr, stepd) => Zipper(id, metadata, stepps, curr, done :+ stepd)
+                (curr, stepd) => Zipper(id, metadata, stepps, curr, (done :+ stepd) ::: toSkip.map(_.copy(skipped = true)))
               )
           }
+        }
         // Current step ongoing
         case Some(stz) => Some(Zipper(id, metadata, pending, stz, done))
       }
 
     def rollback: Zipper = this.copy(focus = focus.rollback)
 
+    def skips: Option[Zipper] = {
```

Comment:
The steps are normally skipped inside `next`. This function takes care of skipping the leading steps when starting the execution of a sequence.